nntrainer/include/layer.h

```diff
@@ -306,20 +310,10 @@ class Layer {
   void setTrainable(bool train) { trainable = train; }
 
   /**
-   * @brief     get gradients
-   * @retval    shared ptr of vector of all tensors
+   * @brief     get updatable params of all
+   * @retval    vector of all params
    */
-  std::shared_ptr<std::vector<Tensor>> getGradients() {
-    return getObjFromRef(gradients);
-  }
-
-  /**
-   * @brief     get weights
-   * @retval    shared ptr of vector of all tensors
-   */
-  std::shared_ptr<std::vector<Tensor>> getWeights() {
-    return getObjFromRef(weights);
-  }
+  std::shared_ptr<UpdatableParam> getParams() { return params; }
```

Comment:
how about implement getGradients() and getWeights(). Cause user have to handle if he want to get gradient only.

Reply:
Yes, I'll work on that once this is merged. Thanks